src/test/java/hudson/tasks/junit/TestResultTest.java

```diff
@@ -283,6 +286,46 @@ public void testTestcaseWithEmptyName() throws Exception {
         assertEquals("(?)", case1.getName());
     }
 
+    @Test
+    public void skipOldReports() throws Exception {
+        long start = System.currentTimeMillis();
+        File testResultFile1 = new File("src/test/resources/hudson/tasks/junit/old-reports/junit-report-1.xml");
+        Files.setLastModifiedTime(testResultFile1.toPath(), FileTime.fromMillis(start + 10));
+        File testResultFile2 = new File("src/test/resources/hudson/tasks/junit/old-reports/junit-report-2.xml");
+        Files.setLastModifiedTime(testResultFile2.toPath(), FileTime.fromMillis(start - 4000));
```

Comment:
Beware that timestamp modifications on Windows are often quite imprecise.

Reply:
that's why I finally added this https://github.com/jenkinsci/junit-plugin/pull/384/files#diff-d80b11180304e55de1556e79fdbd96a795037aeca3c3817cc52aa5734e966b0eR122